### PR TITLE
[bump major] Completely rewrote the log service

### PR DESCRIPTION
**Pull request by jer**
- Splits logs up into fixed-size chunks (100 lines per file)
- Uploads to the Screwdriver Store
- Limits line length to 1000 characters

### CODE REVIEW

**Comment by jer** [comment]
Still left to do:
- [x] Commit the test for SDStoreUpload (WIP)
- [x] Add a flag for line limit so it doesn't have to be hardcoded here and in the api
- [ ] Retries on file upload

File: logservice.go [hunk 1]
```diff
@@ -3,69 +3,141 @@ package main
...
+}
+
+// Uploader returns an Uploader object for the Screwdriver Store
+func (a app) Uploader() sdstoreuploader.SDStoreUploader {
+	return sdstoreuploader.NewFileUploader(a.buildID, a.url, a.token)
+}
 
+// LogReader returns a Reader that is the log source.
+func (a app) LogReader() io.Reader {
 	// If we can't open the socket in the first 60s, the sender probably
```
Comment (petey) [inline logservice.go]: This appears to be 10 minutes.

File: sdstoreuploader/sdstoreuploader.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+// SDError is an error response from the Screwdriver API
+type SDError struct {
+	StatusCode int    `json:"statusCode"`
+	Reason     string `json:"error"`
+	Message    string `json:"message"`
+}
+
+// Error implemetns the error interface for SDError
```
Comment (petey) [inline sdstoreuploader/sdstoreuploader.go]: spelling.

File: sdstoreuploader/sdstoreuploader.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+// handleResponse attempts to parse error objects from Screwdriver
+func handleResponse(res *http.Response) ([]byte, error) {
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("reading response Body from Screwdriver: %v", err)
+	}
+
+	if res.StatusCode/100 != 2 {
```
Comment (petey) [inline sdstoreuploader/sdstoreuploader.go:87]: sneaky

**Review by petey — COMMENTED** [review]
Looks ok to me.

**Review by FenrirUnbound — APPROVED** [review]
👍 

Looks good to me. I can see the payout from investing so much into the tests 😺 

cheeryessentialbass-size_restricted

File: sdstoreuploader/sdstoreuploader.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	req.Header.Set("Authorization", tokenHeader(s.token))
+	req.Header.Set("Content-Type", bodyType)
+	req.ContentLength = size
+
+	res, err := s.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode/100 == 5 {
```
Comment (stjohnjohnson) [inline sdstoreuploader/sdstoreuploader.go:149]: Isn't this use-case already captured by line 87?

**Review by stjohnjohnson — APPROVED** [review]
Looks amazing.  Well thought out.

**Pull request merged**